src/InstrumentationEngine.Lib/LoggerService.cpp

```diff
@@ -67,14 +67,12 @@ HRESULT CLoggerService::GetLoggingFlags(_Out_ LoggingFlags* pLoggingFlags)
 
     IfNotInitRetUnexpected(m_initialize);
 
-    CCriticalSectionHolder holder(&m_cs);
-
     // Intentionally returning the effective flags instead of the default flags
     // (ones set by default or by SetLoggingFlags). The GetLoggingFlags method
     // is primarily meant for consumers external of the Instrumentation Engine
     // assembly to determine which flags are supported rather than literally
     // reporting what the default was or what was set by SetLoggingFlags.
-    *pLoggingFlags = m_effectiveFlags;
+    *pLoggingFlags = (LoggingFlags)m_effectiveFlags.load(std::memory_order_acquire);
```

Comment:
You'll also need to change AllowLogEntry

https://github.com/microsoft/CLRInstrumentationEngine/blob/d034e98ada89a49a08dfc94bbf850c0f301714d1/src/InstrumentationEngine.Lib/LoggerService.cpp#L277

Reply:
Do you mean I should change m_instrumentationMethodFlags to type atomic<LoggingFlags> ?

---
In reply to: [421832010](https://github.com/microsoft/CLRInstrumentationEngine/pull/258#discussion_r421832010) [](ancestors = 421832010)

Reply:
AllowLogEntry is always called within a critical section, there should be no race there.

---
In reply to: [421837264](https://github.com/microsoft/CLRInstrumentationEngine/pull/258#discussion_r421837264) [](ancestors = 421837264,421832010)

Reply:
Though, we might be able to save additional time if AllowLogEntry can be pulled out of the critical sections.

---
In reply to: [421839504](https://github.com/microsoft/CLRInstrumentationEngine/pull/258#discussion_r421839504) [](ancestors = 421839504,421837264,421832010)